lake-primitives/src/types/block.rs

```diff
@@ -79,33 +79,31 @@ impl Block {
         &self.transactions
     }
 
-    pub fn actions(&self) -> Vec<receipts::Action> {
+    pub fn actions_from_streamer_message(&self) -> Vec<receipts::ActionKind> {
         self.streamer_message()
             .shards
             .iter()
             .flat_map(|shard| shard.receipt_execution_outcomes.iter())
             .filter_map(|receipt_execution_outcome| {
-                receipts::Action::try_from(&receipt_execution_outcome.receipt).ok()
+                receipts::ActionKind::try_vec_from_receipt_view(&receipt_execution_outcome.receipt)
+                    .ok()
             })
+            .flatten()
             .collect()
     }
 
-    pub fn events(&mut self) -> Vec<events::Event> {
-        self.receipts()
-            .iter()
-            .flat_map(|executed_receipt| {
-                executed_receipt.logs().into_iter().filter_map(|log| {
-                    if let Ok(event) = events::RawEvent::from_log(&log) {
-                        Some(events::Event {
-                            related_receipt_id: executed_receipt.receipt_id(),
-                            raw_event: event.clone(),
-                        })
-                    } else {
-                        None
-                    }
-                })
-            })
-            .collect()
+    pub fn actions(&mut self) -> std::slice::Iter<'_, impl Action + std::fmt::Debug + Clone> {
```

Comment:
I would return the `ActionKind` enum (see another comment where I suggest to rename it to `Action` enum)